packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -649,6 +655,8 @@ export class GarbageCollector implements IGarbageCollector {
         // Whether we are running in tombstone mode. This is enabled by default if sweep won't run. It can be disabled
         // via feature flags.
         this.tombstoneMode = !this.shouldRunSweep && this.mc.config.getBoolean(disableTombstoneKey) !== true;
+        // Whether we are running in sweep datastores mode.
+        this.sweepDataStoresMode = this.shouldRunSweep && this.mc.config.getBoolean(sweepDatastoresKey) === true;
```

Comment:
Garbage collector should not be checking whether data store sweep is enabled. It should just check if sweep is enabled and act accordingly. Container runtime is the one that should check data store / attachment blob sweep flags and decide what to do.

Reply:
Basically, similar to what we do for tomsbtones. Also, for enabling overall sweep, there is already a flag that is checked when initializaing shouldRunSweep.

Reply:
Sounds good to me, will do